src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/Sdk/Sdk.props

```diff
@@ -0,0 +1,5 @@
+<Project>
+  <PropertyGroup>
+    <MonoAOTCompilerTasksAssemblyPath>$(MSBuildThisFileDirectory)..\tasks\MonoAOTCompiler.dll</MonoAOTCompilerTasksAssemblyPath>
+  </PropertyGroup>
```

Comment:
Related to: https://github.com/dotnet/runtime/issues/47630

Is `Microsoft.NET.Runtime.MonoAOTCompiler.Task` supposed to continue to work as a `@(PackageReference)`? Or is it an MSBuild SDK now?

If this package is an MSBuild SDK now, I think [these docs](https://github.com/dotnet/runtime/blob/171812ba7487283eaf6eff9d7c2bc97f96ef7c15/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md) should be updated showing to do something like this instead:

```xml
<Project Sdk="Microsoft.NET.Sdk">
   <Import Sdk="Microsoft.NET.Runtime.MonoAOTCompiler.Task" Project="Sdk.props" />
   <!-- Rest of the project, that consumes $(MonoAOTCompilerTasksAssemblyPath) -->
</Project>
```

Assuming this one works ^^

Reply:
/cc @akoeplinger

Reply:
We needed this to be an SDK for it to be imported as a pack. But, you could have it be both a regular msbuild package and a SDK by having a `build\$TFM\$PackageName.props` file that imports the SDK (or vice-versa).

Reply:
Ok, so if we use this will it automatically bring in things like the ``ASP.NET Core SDK's`` for when we use ASP.NET Core things too (like dependency injection).

Reply:
One problem is that using this as a `<PackageReference/>` did not actually work as I found in #47630. `<UsingTask>` gets evaluated before the `build\$TFM\$PackageName.props` file sets `$(MonoAOTCompilerTasksAssemblyPath)`...

If both SDK & `@(PackageReference)` could work, it would be good to keep both files. 👍

Reply:
You could move the `UsingTask` inside `$PackageName.targets` / `Sdk.targets`. That way you're not relying on the property evaluation order.

Reply:
``UsingTask`` should always be under ``*.targets`` files anyway.